pypowsybl/rao/impl/rao.py

```diff
@@ -24,45 +23,49 @@ class Rao:
     def __init__(self, handle: _pypowsybl.JavaHandle):
         self._handle = handle
 
-    def set_crac_file_source(self, network: Network, crac_file: Union[str, PathLike]) -> None:
-        crac = io.BytesIO(open(path_to_str(crac_file), "rb").read())
-        self.set_crac_buffer_source(network, crac)
-
-    def set_crac_buffer_source(self, network: Network, crac_source: io.BytesIO) -> None:
-        _pypowsybl.set_crac_source(network._handle, self._handle, crac_source.getbuffer())
-
-    def set_glsk_file_source(self, network: Network, glsk_file: Union[str, PathLike]) -> None:
-        glsk = io.BytesIO(open(path_to_str(glsk_file), "rb").read())
-        self.set_glsk_buffer_source(network, glsk)
-
-    def set_glsk_buffer_source(self, network: Network, glsk_source: io.BytesIO) -> None:
-        _pypowsybl.set_glsk_source(network._handle, self._handle, glsk_source.getbuffer())
-
-    def run(self, network: Network, parameters: Optional[RaoParameters] = None, rao_provider: str = "SearchTreeRao") -> RaoResult:
+    def run(self, network: Network, parameters: Optional[RaoParameters] = None, rao_provider: str = "SearchTreeRao", crac: Optional[Crac] = None, loop_flow_glsk: Optional[Glsk] = None) -> RaoResult:
```

Comment:
Techically, the CRAC should never be None and thus is not optional. I think it would make more sense having it as a mandatory positional parameter like:

```suggestion
    def run(self, crac: Crac, network: Network, parameters: Optional[RaoParameters] = None, rao_provider: str = "SearchTreeRao", loop_flow_glsk: Optional[Glsk] = None) -> RaoResult:
```

Reply:
Good catch, I forgot to update that.